GET.md

```diff
@@ -8,44 +8,67 @@ Read on if you want to setup your own cluster, OpenFaaS and a private GitHub App
 
 * Setup [OpenFaaS](https://github.com/openfaas/faas) and the [faas-cli](https://github.com/openfaas/faas-cli)
 
-* Now get your publically-available URL for OpenFaaS (or one punched out with an ngrok.io tunnel)
+* Now get your publicly-available URL for OpenFaaS (or one punched out with an ngrok.io tunnel)
 
-* Install Derek as a Github app and get your private key, save it as "derek.pem" and put it into the auth folder.
+* Install Derek as a GitHub app and get your private key, save it as "derek.pem".
 
-### Configure Docker image:
-
-We have to build a Docker image with your .pem file included
+* It is also recommended that you set a webhook secret within the GitHub application settings.
 
-We'll also set the symmetric key or secret that you got from GitHub as the `secret_key` environmental variable. Validating via a symmetric key is also known as HMAC. If you want to turn this off (to edit and debug) then set `validate_hmac="false"`
+### Add your secrets :
+  
+Using the method appropriate to the orchestrator chosen during the OpenFaaS setup add `derek.pem` and the GitHub webhook secret as `derek_private_key` and `derek_secret_key` respectively.
 
-Fill out the `application` variable with the ID of the registered Derek Github App, and the `installation` variable with the installation ID you got when adding Derek to your account.
+### Configure Docker image:
 
-Set the following in your Dockerfile
+Most of the configuration is outside of the image, the exception being the version of the OpenFaaS watchdog which is pulled from [GitHub](https://github.com/openfaas/faas/releases); check the desired version is being pulled into the image and go ahead and build:  
 
 ```
-ENV secret_key="docker"
-ENV application=4385
-ENV installation=45362
-ENV private_key="derek.pem"
+$ docker build -t derek .
+```
 
-ENV validate_hmac="true"
+### Configure stack.yml:
+
+This is where Derek finds the details he needs to do the work he does.  The main areas that will need to be updated are the installation and application variables.  The gateway value may also need amending if the gateway is remote.
+
+``` yml
+provider:
+  name: faas
+  gateway: http://localhost:8080  \# can be a remote server
+  
+functions:
+  derek:
+    handler: ./derek
+    image: derek
+    lang: Dockerfile
+  environment:
+    installation: <your_GH_installationID>
```

Comment:
Is `installation` still used in the code?

Reply:
yes, eg `MakeAccessTokenForInstallation(appID string, installation int)`

Reply:
I meant is the environmental variable used? We pull in the installation number from request JSON.

https://github.com/alexellis/derek/blob/7401caeadbca5458bc2d0117ca36b007a82812b2/pullRequestHandler.go#L23